client/client.go

```diff
@@ -10,29 +11,79 @@ import (
 )
 
 func Start() (*ldap.Conn, error) {
-	conf, err := config.Get()
+	conf := config.Get()
+
+	ldapUrl, err := url.Parse(conf.LdapConfig.Address)
 	if err != nil {
-		slog.With("error", err).Debug("Can't get the config")
 		return nil, err
 	}
 
-	ldapUrl, err := url.Parse(conf.LdapConfig.Address)
-	if ldapUrl.Scheme == "ldap" {
-		slog.With("scheme", ldapUrl.Scheme)
-		if conf.LdapConfig.StartTLS {
-			slog.With("useStartTLS", conf.LdapConfig.StartTLS).Debug("Trying to use startTLS")
-			// We need to use startTLS
-		} else {
-			slog.With("useStartTLS", conf.LdapConfig.StartTLS).Debug("The connection will be INSECURE")
+	tlsConfig := tls.Config{InsecureSkipVerify: conf.LdapConfig.InsecureSkipVerify}
+
+	var c *ldap.Conn = nil
+
+	switch ldapUrl.Scheme {
+	case "ldap":
+		{
+			slog.With("scheme", ldapUrl.Scheme)
+			if conf.LdapConfig.StartTLS {
+				slog.With("address", conf.LdapConfig.Address).
+					Debug("Trying to connet to ldap server...")
+				c, err = ldap.DialURL(conf.LdapConfig.Address)
+				if err != nil {
+					return nil, err
+				}
+
+				slog.With("useStartTLS", conf.LdapConfig.StartTLS).
+					Debug("Trying to use startTLS")
+				err = c.StartTLS(&tlsConfig)
+				if err != nil {
+					slog.With("err", err).Debug("StartTLS has failed")
+				}
+
+				if !conf.LdapConfig.InsecureNoSSL {
+					slog.With("InsecureNoSSL", conf.LdapConfig.InsecureNoSSL).
+						Debug("Using ldap without tls. The connection will be INSECURE")
+				} else {
+					return nil, errors.New("Can't establish TLS connection and InsecureSkipVerify is false. The connection can't be secure. Aborting")
+				}
+			} else {
+				slog.With("useStartTLS", conf.LdapConfig.StartTLS).
+					Debug("Using ldap without tls. The connection will be INSECURE")
+				slog.With("address", conf.LdapConfig.Address).
+					Debug("Trying to connet to ldap server...")
+
+				c, err = ldap.DialURL(conf.LdapConfig.Address)
+				if err != nil {
+					return nil, err
+				}
+			}
 		}
-	} else if ldapUrl.Scheme == "ldaps" {
-		slog.With("scheme", ldapUrl.Scheme).Debug("Trying to use ldaps://")
-		// c, err := ldap.DialURL(ldapUrl)
-	} else {
-		err := errors.New("Can't parse a valid scheme in ldap Address: " + ldapUrl.Scheme)
-		slog.With("error", err, "scheme", ldapUrl.Scheme)
-		return nil, err
+	case "ldaps":
+		{
+			slog.With("scheme", ldapUrl.Scheme).Debug("Trying to use ldaps://")
+			slog.With("address", conf.LdapConfig.Address).
+				Debug("Trying to connet to ldap server...")
+
+			c, err = ldap.DialURL(conf.LdapConfig.Address,
+				ldap.DialWithTLSConfig(&tlsConfig))
+			if err != nil {
+				return nil, err
+			}
+		}
+	default:
+		{
+			err := errors.New("Can't parse a valid scheme in ldap Address: " +
```

Comment:
Boh scriverei solo `Invalid scheme in server URI: %s` e non farei la stampa di debug.